onnxruntime/contrib_ops/cuda/transformers/beam_search_topk.cu

```diff
@@ -187,26 +192,14 @@ void LaunchBeamSearchOnlineTopKStage2Kernel(
     cudaStream_t stream) {
   ORT_ENFORCE(parts_per_beam <= 128, "Parts per beam should not be greater than 128");
 
-  int smem_stage2_size = parts_per_beam * max_k * 2 * sizeof(int32_t);
+  int smem_stage2_size = parts_per_beam * K * 2 * sizeof(int32_t);
 
-  if (parts_per_beam <= 32) {
-    BeamSearchOnlineTopKStage2Kernel<T, max_k, 32><<<batch_size * num_beams, 32, smem_stage2_size, stream>>>(
-        topk_values_tmp, topk_indices_tmp, K, vocab_size, parts_per_beam, output_values, output_indices);
-    return;
-  }
-
-  if (parts_per_beam <= 64) {
-    BeamSearchOnlineTopKStage2Kernel<T, max_k, 64><<<batch_size * num_beams, 64, smem_stage2_size, stream>>>(
-        topk_values_tmp, topk_indices_tmp, K, vocab_size, parts_per_beam, output_values, output_indices);
-    return;
-  }
-
-  BeamSearchOnlineTopKStage2Kernel<T, max_k, 128><<<batch_size * num_beams, 128, smem_stage2_size, stream>>>(
+  BeamSearchOnlineTopKStage2Kernel<T, 128><<<batch_size * num_beams, 128, smem_stage2_size, stream>>>(
```

Comment:
any impact on performance after removing 32 and 64 paths?

Reply:
stage 2 is fine. Stage 1 gets much slower. Looking.